pkg/migrations/op_drop_not_null.go

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package migrations
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/lib/pq"
+	"github.com/xataio/pgroll/pkg/schema"
+)
+
+type OpDropNotNull struct {
+	Table  string `json:"table"`
+	Column string `json:"column"`
+	Up     string `json:"up"`
+	Down   string `json:"down"`
+}
+
+var _ Operation = (*OpDropNotNull)(nil)
+
+func (o *OpDropNotNull) Start(ctx context.Context, conn *sql.DB, stateSchema string, s *schema.Schema, cbs ...CallbackFn) error {
+	table := s.GetTable(o.Table)
+	column := table.GetColumn(o.Column)
+
+	// Create a copy of the column on the underlying table.
+	d := NewColumnDuplicator(conn, table, column).WithoutNotNull()
+	if err := d.Duplicate(ctx); err != nil {
+		return fmt.Errorf("failed to duplicate column: %w", err)
+	}
+
+	// Add a trigger to copy values from the old column to the new, rewriting NULL values using the `up` SQL.
```

Comment:
technically it's rewriting all values 😇 it will be NULL ones only if the up function does the typical `CASE`.

Reply:
Thanks, I've updated the comment.